svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -394,62 +394,76 @@ public function removeSemicolons($value)
-            $serverUrl = trim($this->getOption('SlickServerUrl', 'https://app.slickstream.com'));
+            $defaultServerUrl = 'https://app.slickstream.com';
+            $serverUrl = trim($this->getOption('SlickServerUrl', $defaultServerUrl));
             if (substr($serverUrl, 0, 11) === 'adthrive-ab') {
                 $pieces = explode(" ", $serverUrl);
-                $serverUrl = 'https://app.slickstream.com';
+                $serverUrl = $defaultServerUrl;
                 $adThriveAbTest = true;
-                if (count($pieces) > 1) {
-                    $fractionValue = intval($pieces[1]);
-                    if ($fractionValue > 0 && $fractionValue < 100) {
-                        $adThriveAbFraction = $fractionValue / 100;
-                    }
-                }
+                $enabledPercent = (count($pieces) > 1) ? intval($pieces[1]) : 100;
             }
-            echo '<script>' . "\n";
-            echo '"use strict";' . "\n";
+
+            $jsBlock = <<<JSBLOCK
+            window.slickAbTestResult = function(percentEnabled, forceEnable = false, testName = 'On') {
+                const win = window;
+                const storage = win.localStorage;
+                const targetPercentEnabled = parseInt(percentEnabled);
+                
+                if (isNaN(targetPercentEnabled)) {
+                    return new Error("Invalid enabled percentage");
+                }
+
+                let enableSlickFeature = 1;
+
+                const abGroupVal = `slk\${testName}\${targetPercentEnabled}`;
+                const storedOnOffVal = storage.getItem(abGroupVal);
+                const percentKey = `slickAbTestPercent-\${testName}`;
+                const storedPercentVal = parseInt(storage.getItem(percentKey));
+                
+                if (forceEnable !== true || !storedOnOffVal || storedPercentVal !== targetPercentEnabled) {
+                    enableSlickFeature = (Math.random() * 100) <= targetPercentEnabled ? 1 : 0;
+                }
+
+                storage.setItem(abGroupVal, enableSlickFeature);
+                storage.setItem(percentKey, targetPercentEnabled);
+
+                win.adthrive = win.adthrive || {};
+                win.adthrive.cmd = win.adthrive.cmd || [];
+                win.adthrive.cmd.push(() => { win.adthrive.config.abGroup.set(abGroupVal, enableSlickFeature); });
+
+                return !!enableSlickFeature;
+            };
+            JSBLOCK;
+
+            echo "<script>\n";
+            echo "'use strict';\n";
             if ($adThriveAbTest) {
-                echo '(() => {' . "\n";
-                echo '  window.adthrive = window.adthrive || {};' . "\n";
-                echo '  window.adthrive.cmd = window.adthrive.cmd || [];' . "\n";
-                echo '  let slickParams = new URLSearchParams(document.location.search.substring(1));' . "\n";
-                echo '  let slickAbParam = slickParams.get("abEnabled");' . "\n";
```

Comment:
Do we no longer want to support overriding the AB value w/ query string params?

Reply:
Sounds good, I just wasn't sure if the url params were being used by non-tech folks to get around the AB test

Reply:
There's another parameter that does what you're describing but it's not this one. I forget the exact name though.